PowerEditor/src/MISC/Common/Common.cpp

```diff
@@ -52,25 +54,41 @@ generic_string commafyInt(size_t n)
 
 std::string getFileContent(const TCHAR *file2read)
 {
-	if (!::PathFileExists(file2read))
-		return "";
+	if(file2read == nullptr)
+	{
+		return {};
+	}
+	
+	[[maybe_unused]] std::error_code error;
+	if (!std::filesystem::exists(file2read, error))
+	{
+		return {};
+	}
+
+	std::ifstream file(std::filesystem::path(file2read), std::ios::binary);
+	if (!file.is_open()) 
+	{
+		return {};
+	}
+
+	file.seekg(0ul, std::ios::end);
+	const auto fileSize = file.tellg();
+	if (fileSize == 0)
+	{
+		return {};
+	}
+	file.seekg(0ul, std::ios::beg);
 
-	const size_t blockSize = 1024;
-	char data[blockSize];
-	std::string wholeFileContent = "";
-	FILE *fp = _wfopen(file2read, TEXT("rb"));

```

Comment:
Ok, I was just noticing your CONTRIBUTING.md file "Use C++11/14/17/17/20 whenever it is possible." 

Please note,

Also, I made a slight modification to the file-reading logic:

```
size_t lenFile = 0;
do
{
    lenFile = fread(data, 1, blockSize, fp);
    if (lenFile <= 0) break;

```
It may be a potential issue. The value returned from fread cannot be less than 0. Furthermore, since lenFile is an unsigned type, a value less than zero isn't feasible. In the event of a file-reading error, this code might result in an infinite loop and subsequently crash the application. This is because if there's an error, fread can return a positive value that may not match the buffer size.

Thank you.

Reply:
One thing at a time.
This PR is focused on crash issue. So please fix it with minimum fix.
For the other issue link to this function, please create an issue and we will see if we want to treat it.